subprojects/build-cache-packaging/src/main/java/org/gradle/caching/internal/command/BuildCacheCommandFactory.java

```diff
@@ -178,32 +162,14 @@ public OriginMetadata getOriginMetadata() {
         private void cleanLocalState() {
             entity.visitLocalState(localStateFile -> {
                 try {
-                    remove(localStateFile);
+                    if (localStateFile.exists()) {
+                        FileUtils.forceDelete(localStateFile);
```

Comment:
We are now removing the directory as well, and not only cleaning it. Is that expected for `localState`?

Reply:
Moved all the behavior to `CacheStep` now, and are removing things the same way as we did before.